File: data/src/main/java/com/github/anrimian/musicplayer/data/storage/providers/playlists/StoragePlayList.java
package com.github.anrimian.musicplayer.data.storage.providers.playlists;

import androidx.annotation.NonNull;

import java.util.Date;

import javax.annotation.Nonnull;

public class StoragePlayList {

    private long storageId;

    @Nonnull
    private String name;

    @Nonnull
    private Date dateAdded;

    @Nonnull
    private Date dateModified;

    public StoragePlayList(long storageId,
                           @Nonnull String name,
                           @Nonnull Date dateAdded,
                           @Nonnull Date dateModified) {
        this.storageId = storageId;
        this.name = name;
        this.dateAdded = dateAdded;
        this.dateModified = dateModified;
    }

    public long getStorageId() {
        return storageId;
    }

    @Nonnull
    public String getName() {
        return name;
    }

    @Nonnull
    public Date getDateAdded() {
        return dateAdded;
    }

    @Nonnull
    public Date getDateModified() {
        return dateModified;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        StoragePlayList playList = (StoragePlayList) o;

        return storageId == playList.storageId;
    }

    @Override
    public int hashCode() {
        return (int) (storageId ^ (storageId >>> 32));
    }

    @NonNull
    @Override
    public String toString() {
        return "PlayList{" +
                "id=" + storageId +
                ", name='" + name + '\'' +
                ", dateAdded=" + dateAdded +
                ", dateModified=" + dateModified +
                '}';
    }
}

File: data/src/main/java/com/github/anrimian/musicplayer/data/storage/providers/playlists/StoragePlaylist.kt
package com.github.anrimian.musicplayer.data.storage.providers.playlists

class StoragePlaylist(
    val storageId: Long,
    val name: String,
    val addedTime: Long,
    val modifiedTime: Long
) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is StoragePlaylist) return false

        if (storageId != other.storageId) return false

        return true
    }

    override fun hashCode(): Int {
        return storageId.hashCode()
    }

    override fun toString(): String {
        return "StoragePlayList(storageId=$storageId, name='$name', addedTime=$addedTime, modifiedTime=$modifiedTime)"
    }

}
